--- a/ggml/src/ggml-cuda/mmvq.cu
+++ b/ggml/src/ggml-cuda/mmvq.cu
@@ -449,12 +449,14 @@
                 case GGML_TYPE_Q5_1:
                 case GGML_TYPE_Q8_0:
                 case GGML_TYPE_Q2_K:
-                case GGML_TYPE_Q4_K:
                 case GGML_TYPE_Q5_K:
-                case GGML_TYPE_Q6_K:
                 case GGML_TYPE_IQ4_NL:
                 case GGML_TYPE_IQ4_XS:
                     return 8;
+                case GGML_TYPE_Q4_K:
+                    return 3;
+                case GGML_TYPE_Q6_K:
+                    return 5;
                 default:
                     return 1;
             }
